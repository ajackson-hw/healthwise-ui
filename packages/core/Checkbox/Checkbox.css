.checkbox_wrapper {
  position: relative;
  display: block;
}

.checkbox {
  position: absolute;
  display: block;
  width: 1em;
  height: 1em;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
  cursor: pointer;
  font-size: inherit;
}

.checkbox:disabled {
  cursor: not-allowed;
}

.icon {
  display: block;
  width: 1em;
  height: 1em;
}

.checkbox:focus ~ .icon {
  outline: var(--focus-indicator);
  outline-offset: var(--focus-indicator-offset);
}

.checkbox:disabled ~ .icon {
  opacity: 0.35;
  pointer-events: none;
}

.icon svg {
  fill: var(--color-text-primary);
}

.checkbox:invalid ~ .icon svg {
  fill: var(--color-error);
}

.checked {
  display: none;
}

.unchecked {
  display: block;
}

.checkbox:checked ~ .checked {
  display: block;
}

.checkbox:checked ~ .unchecked {
  display: none;
}
