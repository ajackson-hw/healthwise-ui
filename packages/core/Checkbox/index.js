import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styled from 'styled-components'

import { defaultTheme } from '../Theme'
import { CheckboxCheckedIcon, CheckboxUncheckedIcon } from '../Icon'

const Root = styled.div`
  position: relative;
  display: block;
`

const CheckedIconContainer = styled.div`
  display: none;
  width: 1em;
  height: 1em;
  svg {
    fill: ${props => props.theme.colorTextPrimary};
  }
`

CheckedIconContainer.defaultProps = {
  theme: defaultTheme,
}

const UncheckedIconContainer = styled.div`
  display: none;
  width: 1em;
  height: 1em;
  svg {
    fill: ${props => props.theme.colorTextPrimary};
  }
`

UncheckedIconContainer.defaultProps = {
  theme: defaultTheme,
}

const Input = styled.input`
  position: absolute;
  display: block;
  width: 1em;
  height: 1em;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
  cursor: pointer;
  font-size: inherit;

  :focus ~ ${CheckedIconContainer},
  :focus ~ ${UncheckedIconContainer} {
    outline: ${props => props.theme.focusIndicator};
    outline-offset: ${props => props.theme.focusIndicatorOffset};
  }

  :disabled {
    cursor: not-allowed;
  }

  :disabled ~ ${CheckedIconContainer},
  :disabled ~ ${UncheckedIconContainer} {
    opacity: 0.35;
    pointer-events: none;
  }

  :invalid ~ ${CheckedIconContainer} svg,
  :invalid ~ ${UncheckedIconContainer} svg {
    fill: ${props => props.theme.colorError};
  }

  :checked ~ ${CheckedIconContainer} {
    display: block;
  }

  :not(:checked) ~ ${UncheckedIconContainer} {
    display: block;
  }
`

Input.defaultProps = {
  theme: defaultTheme
}

class Checkbox extends React.Component {
  render() {
    const { className, name, value, checked, onClick, required, theme, ...otherProps } = this.props

    return (
      <Root className="hw-checkbox-wrapper">
        <Input
          type="checkbox"
          name={name}
          value={value}
          checked={checked}
          required={required}
          className={classNames('hw-checkbox', className)}
          onClick={onClick}
          theme={theme}
          {...otherProps}
        />
        <CheckedIconContainer>
          <CheckboxCheckedIcon role="presentation" />
        </CheckedIconContainer>
        <UncheckedIconContainer>
          <CheckboxUncheckedIcon role="presentation" />
        </UncheckedIconContainer>
      </Root>
    )
  }
}

Checkbox.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onClick: PropTypes.func,
  required: PropTypes.bool,
  theme: PropTypes.shape({
    colorTextPrimary: PropTypes.string,
    focusIndicator: PropTypes.string,
    focusIndicatorOffset: PropTypes.string,
    colorError: PropTypes.string,
  })
}

export default Checkbox
