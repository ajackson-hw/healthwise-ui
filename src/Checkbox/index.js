import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styled, { withTheme } from 'styled-components'

import { defaultTheme } from '../Theme'
import { CheckboxCheckedIcon, CheckboxUncheckedIcon } from '../Icon'

const Root = styled.div`
  position: relative;
  display: block;
`

const CheckedIconContainer = styled.div`
  display: none;
  width: 1em;
  height: 1em;
  svg {
    fill: ${props => props.theme.colorTextPrimary};
  }
`

const UncheckedIconContainer = styled.div`
  display: none;
  width: 1em;
  height: 1em;
  svg {
    fill: ${props => props.theme.colorTextPrimary};
  }
`

const Input = styled.input`
  position: absolute;
  display: block;
  width: 1em;
  height: 1em;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
  cursor: pointer;
  font-size: inherit;

  :focus ~ ${CheckedIconContainer}, :focus ~ ${UncheckedIconContainer} {
    outline: ${props => props.theme.focusIndicator};
    outline-offset: ${props => props.theme.focusIndicatorOffset};
  }

  :disabled {
    cursor: not-allowed;
  }

  :disabled ~ ${CheckedIconContainer}, :disabled ~ ${UncheckedIconContainer} {
    opacity: 0.35;
    pointer-events: none;
  }

  :invalid ~ ${CheckedIconContainer} svg,
  :invalid ~ ${UncheckedIconContainer} svg {
    fill: ${props => props.theme.colorError};
  }

  :checked ~ ${CheckedIconContainer} {
    display: block;
  }

  :not(:checked) ~ ${UncheckedIconContainer} {
    display: block;
  }
`

class Checkbox extends React.Component {
  render() {
    const { className, name, value, checked, onClick, required, theme, ...otherProps } = this.props

    return (
      <Root className="hw-checkbox-wrapper">
        <Input
          type="checkbox"
          name={name}
          value={value}
          checked={checked}
          required={required}
          className={classNames('hw-checkbox', className)}
          onClick={onClick}
          theme={theme}
          {...otherProps}
        />
        <CheckedIconContainer theme={theme}>
          <CheckboxCheckedIcon role="presentation" />
        </CheckedIconContainer>
        <UncheckedIconContainer theme={theme}>
          <CheckboxUncheckedIcon role="presentation" />
        </UncheckedIconContainer>
      </Root>
    )
  }
}

Checkbox.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  checked: PropTypes.bool,
  onClick: PropTypes.func,
  required: PropTypes.bool,
  theme: PropTypes.shape({
    colorTextPrimary: PropTypes.string,
    focusIndicator: PropTypes.string,
    focusIndicatorOffset: PropTypes.string,
    colorError: PropTypes.string,
  }),
}

Checkbox.defaultProps = {
  theme: defaultTheme,
}

export default withTheme(Checkbox)
