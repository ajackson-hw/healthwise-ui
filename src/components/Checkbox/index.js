import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { CheckboxCheckedIcon, CheckboxUncheckedIcon } from '@healthwise/icon'
import styles from './Checkbox.css'

class Checkbox extends React.Component {
  render() {
    const { className, name, value, checked, onClick, required, ...otherProps } = this.props

    const checkboxClassName = classNames('hw-checkbox', styles.checkbox, className)

    return (
      <div className={`hw-checkbox-wrapper ${styles.checkbox_wrapper}`}>
        <input
          type="checkbox"
          name={name}
          value={value}
          checked={checked}
          required={required}
          className={checkboxClassName}
          onClick={onClick}
          {...otherProps}
        />
        <div className={`${styles.icon} ${styles.checked}`}>
          <CheckboxCheckedIcon role="presentation" />
        </div>
        <div className={`${styles.icon} ${styles.unchecked}`}>
          <CheckboxUncheckedIcon role="presentation" />
        </div>
      </div>
    )
  }
}

Checkbox.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onClick: PropTypes.func,
  required: PropTypes.bool,
}

export default Checkbox
