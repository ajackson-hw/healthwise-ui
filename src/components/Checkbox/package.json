{
  "name": "@healthwise/checkbox",
  "version": "3.0.0",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - Checkbox input component",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@healthwise/icon": "^4.0.0",
    "classnames": "^2.2.5",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "@healthwise/theme": "^4.0.0",
    "react": ">=15.4.2"
  }
}
