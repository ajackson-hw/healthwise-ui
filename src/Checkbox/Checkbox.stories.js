import React from 'react'
import { storiesOf } from '@storybook/react'

import Checkbox from './index'

const styles = {
  container: {
    fontSize: '24px',
    marginBottom: '16px',
  },
}

storiesOf('core|Form Controls/Checkbox', module)
  .add(
    'with defaults',
    () => (
      <div>
        <div style={styles.container}>
          <Checkbox name="test" value="test" />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 2" value="test" defaultChecked />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 3" value="test" label="This is a label" />
        </div>
      </div>
    ),
    {
      info: `Demonstrates default rendering of Checkbox component`,
    }
  )
  .add(
    'disabled',
    () => (
      <div>
        <div style={styles.container}>
          <Checkbox name="test" value="test" disabled />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 2" value="test" disabled defaultChecked />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 3" value="test" label="This is a label" disabled />
        </div>
      </div>
    ),
    {
      info: `Demonstrates a disabled Checkbox component`,
    }
  )
  .add(
    'required',
    () => (
      <div>
        <div style={styles.container}>
          <Checkbox name="test" value="test" required />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 2" value="test" required defaultChecked />
        </div>
        <div style={styles.container}>
          <Checkbox name="test 3" value="test" label="This is a label" required />
        </div>
      </div>
    ),
    {
      info: `Demonstrates a required Checkbox component`,
    }
  )
