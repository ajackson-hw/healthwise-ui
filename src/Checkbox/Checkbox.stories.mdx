import { Meta, Story, Preview} from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'

import Checkbox from './index'

<Meta title="Form Controls/Checkbox" component={Checkbox} />

# Checkbox

default

<Preview>
  <Story name="default">
    <Checkbox name="test 1" value="test" />
    <Checkbox name="test 2" value="test" defaultChecked />
    <Checkbox name="test 3" value="test" label="This is a label" />
  </Story>
</Preview>

disabled

<Preview>
  <Story name="disabled">
    <Checkbox name="test 1" value="test" disabled />
    <Checkbox name="test 2" value="test" defaultChecked disabled />
    <Checkbox name="test 3" value="test" label="This is a label" disabled />
  </Story>
</Preview>

viewOnly

<Preview>
  <Story name="viewOnly">
    <Checkbox name="test 1" value="test" label="Label 1" viewOnly />
    <Checkbox name="test 2" value="test" label="Label 2" defaultChecked viewOnly />
    <Checkbox name="test 3" value="test" label="Label 3" viewOnly />
  </Story>
</Preview>

required

<Preview>
  <Story name="required">
    <Checkbox name="test 1" value="test" label="Label 1" required />
    <Checkbox name="test 2" value="test" label="Label 2" defaultChecked required />
    <Checkbox name="test 3" value="test" label="Label 3" required />
  </Story>
</Preview>
