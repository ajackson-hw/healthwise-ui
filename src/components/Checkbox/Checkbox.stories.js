import React from 'react'
import { storiesOf } from '@storybook/react'

import Checkbox from './index'

const styles = {
  container: {
    fontSize: '24px',
    marginBottom: '16px',
  },
}

storiesOf('Form Controls/Checkbox', module)
  .addWithInfo('with defaults', `Demonstrates default rendering of Checkbox component`, () => (
    <div>
      <div style={styles.container}>
        <Checkbox name="test" value="test" />
      </div>
      <div style={styles.container}>
        <Checkbox name="test 2" value="test" defaultChecked />
      </div>
    </div>
  ))
  .addWithInfo('disabled', `Demonstrates a disabled Checkbox component`, () => (
    <div>
      <div style={styles.container}>
        <Checkbox name="test" value="test" disabled />
      </div>
      <div style={styles.container}>
        <Checkbox name="test 2" value="test" disabled defaultChecked />
      </div>
    </div>
  ))
  .addWithInfo('required', `Demonstrates a required Checkbox component`, () => (
    <div>
      <div style={styles.container}>
        <Checkbox name="test" value="test" required />
      </div>
      <div style={styles.container}>
        <Checkbox name="test 2" value="test" required defaultChecked />
      </div>
    </div>
  ))
