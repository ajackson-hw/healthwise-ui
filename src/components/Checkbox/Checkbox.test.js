/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Checkbox from './index'

describe('<Checkbox />', () => {
  it('renders correctly with defaults', () => {
    const component = renderer.create(<Checkbox name="test" value="test value" />)
    expect(component).toMatchSnapshot()
  })

  it('renders checked checkbox correclty', () => {
    const component = renderer.create(<Checkbox name="test" value="test value" checked />)
    expect(component).toMatchSnapshot()
  })
})
