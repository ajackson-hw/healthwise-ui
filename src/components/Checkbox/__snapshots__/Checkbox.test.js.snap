// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders checked checkbox correclty 1`] = `
<div
  className="hw-checkbox-wrapper checkbox_wrapper"
>
  <input
    checked={true}
    className="hw-checkbox checkbox"
    name="test"
    onClick={undefined}
    required={undefined}
    type="checkbox"
    value="test value"
  />
  <div
    className="icon checked"
  >
    <div
      className="hw-icon Icon"
    >
      <svg
        className="IconSvg"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zM10.8 17.4l-2.1 2.1-6.2-6.2 2.1-2.1 4.1 4.1L19.4 4.5l2.1 2.1-10.7 10.8z"
        />
      </svg>
    </div>
  </div>
  <div
    className="icon unchecked"
  >
    <div
      className="hw-icon Icon"
    >
      <svg
        className="IconSvg"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21.3 2.7v18.7H2.7V2.7h18.6zm0-2.7H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> renders correctly with defaults 1`] = `
<div
  className="hw-checkbox-wrapper checkbox_wrapper"
>
  <input
    checked={undefined}
    className="hw-checkbox checkbox"
    name="test"
    onClick={undefined}
    required={undefined}
    type="checkbox"
    value="test value"
  />
  <div
    className="icon checked"
  >
    <div
      className="hw-icon Icon"
    >
      <svg
        className="IconSvg"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zM10.8 17.4l-2.1 2.1-6.2-6.2 2.1-2.1 4.1 4.1L19.4 4.5l2.1 2.1-10.7 10.8z"
        />
      </svg>
    </div>
  </div>
  <div
    className="icon unchecked"
  >
    <div
      className="hw-icon Icon"
    >
      <svg
        className="IconSvg"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21.3 2.7v18.7H2.7V2.7h18.6zm0-2.7H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0z"
        />
      </svg>
    </div>
  </div>
</div>
`;
